Automattic/wp-calypso
# StepHeader: move shared component out of signup

StepHeader will be required by two modules, jetpack-connect and signup (#14321). This PR moves it to `blocks`.

To test:
* Use this branch
* Follow flows using `StepHeader`
  * http://calypso.localhost:3000/jetpack/connect/store
  * http://calypso.localhost:3000/start
* Ensure code compiles correctly, tests pass, and there are no regressions or console errors/warnings
